config/has_is_constant_evaluated.cpp

```diff
@@ -3,8 +3,44 @@
 //  Boost Software License, Version 1.0. (See accompanying file
 //  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
+// This program determines if std::is_constant_evaluated  is available to switch within a function to use compile-time constexp calculations.
```

Comment:
I'm not sure about these changes - the purpose of this configuration test is to determine whether BOOST_MP_HAS_IS_CONSTANT_EVALUATED has been set or not.

* There are some compilers (gcc-6/7/8) which do support constexpr detection through means other than `std::is_constant_evaluated`, and in compilation modes other than C++2a.
* Likewise `__builtin_is_constant_evaluated` is available in C++14 and later once it's been added, so some of these checks just create unnecessary warnings?
* There are some compilers (Intel - support for which I'm still working on), which does support `__builtin_is_constant_evaluated` but which is too buggy to use it's C++14 constexpr support at all.  So again IMO this just generates unnecessary warning noise?